File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/converter/CustomerSupportContactConverter.java
package com.worldventures.dreamtrips.wallet.domain.converter;

import com.worldventures.dreamtrips.modules.mapping.converter.Converter;
import com.worldventures.dreamtrips.wallet.domain.entity.settings.customer_support.Contact;
import com.worldventures.dreamtrips.wallet.domain.entity.settings.customer_support.ImmutableContact;

import io.techery.mappery.MapperyContext;

public class CustomerSupportContactConverter implements Converter<com.worldventures.dreamtrips.api.smart_card.documents.customer_support.model.Contact, Contact> {

   @Override
   public Class<com.worldventures.dreamtrips.api.smart_card.documents.customer_support.model.Contact> sourceClass() {
      return com.worldventures.dreamtrips.api.smart_card.documents.customer_support.model.Contact.class;
   }

   @Override
   public Class<Contact> targetClass() {
      return Contact.class;
   }

   @Override
   public Contact convert(MapperyContext mapperyContext, com.worldventures.dreamtrips.api.smart_card.documents.customer_support.model.Contact contact) {
      return ImmutableContact.builder()
            .contactAddress(contact.contactAddress())
            .email(contact.email())
            .fax(contact.fax())
            .formattedAddress(contact.formattedAddress())
            .phone(contact.phone())
            .build();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/converter/CustomerSupportContactConverter.kt
package com.worldventures.wallet.domain.converter

import com.worldventures.core.converter.Converter
import com.worldventures.wallet.domain.entity.settings.customer_support.Contact
import io.techery.mappery.MapperyContext

typealias ApiContact = com.worldventures.dreamtrips.api.smart_card.documents.customer_support.model.Contact

class CustomerSupportContactConverter : Converter<ApiContact, Contact> {

   override fun sourceClass() = ApiContact::class.java

   override fun targetClass() = Contact::class.java

   override fun convert(context: MapperyContext, source: ApiContact) =
         Contact(
               contactAddress = source.contactAddress(),
               email = source.email(),
               fax = source.fax(),
               formattedAddress = source.formattedAddress(),
               phone = source.phone()
         )
}